### PR TITLE
feat(profiling): add zoomIntoFrame strategy

### PR DESCRIPTION
**Pull request by JonasBa**
The behavior to always zoom directly into a frame on certain interactions is confusing and often causes context loss (think of zooming directly into a tiny frame which is already in the view). For certain interactions like running through searches, a min move approach is favorable so that we can preserve as much context around the matched frames as possible and instead just highlight the frame. This PR adds a zoomIntoFrame strategy parameter that will enable moving using min moves. 

The highlighting needs to be fixed, but will come in a different PR (we currently implicitly connect selected node with highlighted node and it needs to be decoupled first)

### CODE REVIEW

File: static/app/utils/profiling/gl/utils.ts [hunk 1]
```diff
@@ -624,3 +641,52 @@ export function computeHighlightedBounds(
...
+): Rect {
+  if (strategy === 'exact') {
+    return frame.withHeight(view.height);
+  }
+
+  if (strategy === 'min') {
+    if (view.width < frame.width) {
+      // If view width < frame width, we need to zoom out,
+      // so the condition = exact
+      return frame.withHeight(view.width);
```
Comment (Zylphrex) [inline static/app/utils/profiling/gl/utils.ts:658]: why is this setting the height to the width?

File: static/app/utils/profiling/gl/utils.ts [hunk 2]
```diff
@@ -624,3 +641,52 @@ export function computeHighlightedBounds(
...
+    if (frame.left < view.left) {
+      // If frame is to the left of the view, translate it left
+      // to frame.x so that start of the frame is in the view
+      offset = offset.withX(frame.x);
+    }
+    if (offset.overlapsRight(frame) || frame.right > view.right) {
+      // If frame is to the right of the view, translate it right
+      // by the difference between the right edge of the frame and the view
+      offset = view.withX(offset.x + frame.right - offset.right);
+    }
```
Comment (Zylphrex) [inline static/app/utils/profiling/gl/utils.ts:676]: Can you explain what the intent here is? I'm not sure I understand what this does.

File: static/app/utils/profiling/gl/utils.ts (hunk 1 above)
Comment (JonasBa) [reply static/app/utils/profiling/gl/utils.ts:658]: It's a bug, I fixed it in the follow up PR, but forgot to fix it here too

File: static/app/utils/profiling/gl/utils.ts [hunk 3]
```diff
@@ -624,3 +641,52 @@ export function computeHighlightedBounds(
 
   return null;
 }
+
+export function computeConfigViewWithStategy(
+  strategy: 'min' | 'exact',
```
[the remark this answers is not in the document]
Comment (JonasBa) [reply static/app/utils/profiling/gl/utils.ts:646]: For sure

File: static/app/utils/profiling/gl/utils.ts [hunk 4]
```diff
@@ -624,3 +641,52 @@ export function computeHighlightedBounds(
...
+  if (strategy === 'exact') {
+    return frame.withHeight(view.height);
+  }
+
+  if (strategy === 'min') {
+    if (view.width < frame.width) {
+      // If view width < frame width, we need to zoom out,
+      // so the condition = exact
+      return frame.withHeight(view.height);
+    }
```
[the remark this answers is not in the document]
Comment (JonasBa) [reply static/app/utils/profiling/gl/utils.ts:659]: Yup, good eye

File: static/app/utils/profiling/gl/utils.ts [hunk 5]
```diff
@@ -624,3 +641,52 @@ export function computeHighlightedBounds(
...
+
+    if (frame.bottom < view.top) {
+      // If frame is above the view, translate view to top of frame
+      offset = offset.withY(frame.top);
+    }
+    if (frame.bottom > view.bottom) {
+      // If frame is below the view, translate view by the difference
+      // of the bottom edge of the frame and the view
+      offset = offset.translateY(offset.y + frame.bottom - offset.bottom);
+    }
```
[the remark this answers is not in the document]
Comment (JonasBa) [reply static/app/utils/profiling/gl/utils.ts:686]: Yeah, agree it makes it more readable. I didnt bother adding them since they werent firing anyways

**Review by Zylphrex — APPROVED** [review]
:shipit:

**Review by eliashussary — APPROVED** [review]

**Pull request merged**